M/MLX/build_tarballs.jl

```diff
@@ -103,7 +102,11 @@ products = Product[
 dependencies = [
     Dependency("libblastrampoline_jll"; compat="5.4", platforms = libblastrampoline_platforms),
     Dependency("OpenBLAS32_jll"; platforms = openblas_platforms),
-    Dependency("OpenMPI_jll"),
+
+    # OpenMPI 5 is ABI compatible with OpenMPI 4, but OpenMPI 5 does not support 32-bit platforms, or FreeBSD
+    Dependency("OpenMPI_jll", v"4.1.8"; compat="4, 5", platforms = filter(p -> nbits(p) == 32 || Sys.isfreebsd(p), platforms)),
+    Dependency("OpenMPI_jll", v"5.0.7"; compat="4, 5", platforms = filter(p -> !(nbits(p) == 32 || Sys.isfreebsd(p)), platforms)),
```

Comment:
```suggestion
    Dependency("OpenMPI_jll"; compat="4.1.8, 5"),
```

Reply:
I've updated the recipe to use
```
    Dependency("OpenMPI_jll", v"4.1.8"; compat="4, 5"), # OpenMPI 5 is ABI compatible with OpenMPI 4
```

To include compat with earlier OpenMPI 4 versions (4.1.8 (and 5.0.7) is/was just a randomly selected (most recent) v4 version to be used for build).

Reply:
> To include compat with earlier OpenMPI 4 versions

But why if it won't work?

Reply:
The only issue is that there is no OpenMPI v4.0.0 - there are plenty of versions prior to v4.1.8, e.g. v4.0.2 I assume should be fine as well (and I guess v4.0.0 would be OK too if someone updated the OpenMPI recipe to push a v4.0.0).

Reply:
> I assume should be fine as well

I always find it extremely dodgy having minimum compat bound != build version (reason why I made the former the default build version): if you have to tell that a package is compatible with ***minimum*** a specific version, what's the point of using an ***earlier*** build version?  On point here, it won't work on riscv64 where there's no build before 4.1.8, so no, it won't be fine as well.

Reply:
OK, I get your point regarding ensuring minimum compat bound == build_version - to have some form of check that the minimum compat actually works.

I had not checked that earlier versions were missing platforms (that are now default).

But on the other hand, compat 4.1.8 is an artificial compat bound - limiting users to using only 4.1.8+ - at the benefit of having a BB build/audit pass.

I have no upstream info on which OpenMPI version is required - the build system is only saying the OpenMPI version is version 3.1 - the only info I had on compat was the comment in the OpenMPI@4 recipe saying that v4 and v5 are ABI compatible.

But I’m fine with limiting to 4.1.8+.

Reply:
I've committed the suggested change.